packages/@aws-cdk/aws-codestarnotifications/lib/source.ts

```diff
@@ -0,0 +1,28 @@
+import { INotificationRule } from './notification-rule';
+
+/**
+ * Information about the Codebuild or CodePipeline associated with a notification source.
+ */
+export interface NotificationRuleSourceConfig {
+  /**
+   * The source type. Can be an AWS CodeCommit, CodeBuild, CodePipeline or CodeDeploy.
+   */
+  readonly sourceType: string;
+
+  /**
+   * The Amazon Resource Name (ARN) of the notification source.
+   */
+  readonly sourceArn: string;
+}
+
+/**
+ * Represents a notification source
+ * The source that allows CodeBuild and CodePipeline to associate with this rule.
+ */
+export interface INotificationRuleSource {
+  /**
+   * Binds source to notification rule
+   * @param _rule The notification rule
+   */
+  bind(_rule: INotificationRule): NotificationRuleSourceConfig;
```

Comment:
1. This should be called something more distinctive, so that it doesn't clash with other `bind()` methods. Something like `bindAsNotificationRuleSource()`.
2. Make it also take a `scope: Construct` as the first argument.